### ISSUE TITLE
Feature Request: Choose Account Contract used for Preloaded Accounts

### ISSUE THREAD

**Issue by rzmahmood** [Shard-Labs/starknet-devnet#285]
Hey team. As we go through more upgrades and Account Contracts evolve, it's become evident that allowing us to choose which Account contract implementation used for the preloaded accounts would be beneficial (or at least the version of OZ used).

For example, now Pathfinder doesn't support Accounts post v0.10.0 due to this issues https://github.com/eqlabs/pathfinder/issues/632

Our testing therefore needs to be done with older accounts, but because we can't choose the Account used with the devnet, some friction is introduced.

**Comment by FabijanC** [comment]
Makes sense. What we need to keep in mind here is that different implementations have different ABIs.

https://github.com/Shard-Labs/starknet-devnet/blob/c7fc832d3454e70196070735a9f6e5cb875fd082/starknet_devnet/account.py#L75

For example, in the above linked section of `account.py`, Devnet depends on the fact that the public key is stored under `Account_public_key`. We don't know if it's going to be under this name and if that is going to be the only parameter requiring setting.

So what could be done is to allow users to specify via a CLI parameter the path to the compiled account JSON artifact and assume that it only needs the public key and that it is under `Account_public_key`. I think this would allow predeploying both the previous (0.3.1) and the current (0.4.0b) OpenZeppelin version.

Or, the public key location could also be a CLI parameter.
